### PR TITLE
Remove react tabs props

### PR DESCRIPTION
**Pull request by abarciauskas-bgse**
Removes `onSelect` and `selectedIndex` property from react tabs.

**Summary:** After noticing warnings from tests ([see this build for example](https://travis-ci.org/CMSgov/qpp-submissions-docs/builds/269599921)), some pairing with @samskeller and further investigation, we realized that the `onSelect` property was undefined on `<Tabs />` properties. I don't think we need the `onSelect` or `selectedIndex` properties at all - they are required for controlled mode of react tabs but uncontrolled mode appears to work fine for our app.

**Investigation:**

Rendering tree: `<AdvancedTutorial />` or `<BasicTutorial />` call `<TechnicalDetailsPane />`'s which calls `<Basic1-3 />` and `<Advanced1-5 />` which all call `<Tabs />`.

All `<Tabs />` components are instantiated with the following properties in the tutorials:

```
<Tabs
  className={cssClass}
  onSelect={this.props.onSelect}
  selectedIndex={this.props.tabIndex}>
```

See `grep -r "<Tabs" src/components/tutorials/common/steps -A 3`

[technical-details-pane.jsx](https://github.com/CMSgov/qpp-submissions-docs/blob/master/src/components/tutorials/common/technical-details-pane.jsx) renders `<Basic1-3 />` and `<Advanced1-5 />` dependent on the `tutorial` and `hash` properties of the `<TechnicalDetailsPane />`, for example:

```
<TechnicalDetailsPane
  tutorial="basic"
  hash="#creating-a-submission"
  tabIndex={this.state.tabIndex}
  selectTab={this.selectTab} />
```

In the example above, `<Basic1 onSelect={selectTab} tabIndex={index}/>` is rendered.

Calls to `<TechnicalDetailsPane />`'s set _either_ the:
* `selectTab` property to `this.selectTab` in `basic-tutorial`, or,
* `handleSelect` property to `this.handleSelect` (in `advanced-tutorial`).

[`selectTab` _is_ defined in `basic-tutorial`](https://github.com/CMSgov/qpp-submissions-docs/blob/master/src/components/tutorials/basic-tutorial.jsx#L19-L23) however `handleSelect` is not defined in `advanced-tutorial`. 

See
* `grep -r "<TechnicalDetailsPane" src/components/tutorials/ -A 4`
* [advanced-tutorial.jsx](https://github.com/CMSgov/qpp-submissions-docs/blob/9d62b9b87ba6308ba270403b2b2736e5843a3eb1/src/components/tutorials/advanced-tutorial.jsx)
* [basic-tutorial.jsx](https://github.com/CMSgov/qpp-submissions-docs/blob/master/src/components/tutorials/basic-tutorial.jsx)

[technical-details-pane.jsx](https://github.com/CMSgov/qpp-submissions-docs/blob/master/src/components/tutorials/common/technical-details-pane.jsx) always sets `onSelect={selectTab}`. From this I deduced `onSelect` _is_ defined for `<Tabs />` when `<BasicTutorial />` is rendered but _is not_ defined for `<Tabs />` when `<AdvancedTutorial />` is rendered.

This is because `onSelect` in `<Tabs />` is set to `this.props.onSelect` which is set to `this.props.selectTab` in `<TechnicalDetailsPane />` and `selectTab` _is_ defined in `basic-tutorial` but not in `advanced-tutorial`.

I verified this by loading the `/qpp-submissions-docs/tutorial` and `/qpp-submissions-docs/advanced-tutorial` pages and inspecting the errors logged in the console. There are no errors when you load `/qpp-submissions-docs/tutorial` but errors for all components arise when you load `/qpp-submissions-docs/advanced-tutorial`. This is also apparent from the warning stack trace:

```
console.error node_modules/fbjs/lib/warning.js:36
    Warning: Failed prop type: The prop `onSelect` is marked as required in `Basic1`, but its value is `undefined`.
        in Basic1 (at technical-details-pane.jsx:22)
        in TechnicalDetailsPane (at advanced-tutorial.jsx:103)
        in div (at advanced-tutorial.jsx:102)
        in div (at advanced-tutorial.jsx:54)
        in div (at advanced-tutorial.jsx:53)
        in div (at advanced-tutorial.jsx:46)
        in AdvancedTutorial (created by Route)
```

**Testing:**
* Adhoc testing in the UI

**Post-script**
* I think this actually fixes a bug: On current master if you load `/qpp-submissions-docs/tutorial` and change tabs for "Creating a new submission" it changes tabs for "Adding measurements". I think this is because we are setting `onSelect` to a handler at the `basic-tutorial` level and not at the <Tabs /> level. This is not the case for advanced-tutorial because advanced-tutorial is already implicitly using uncontrolled mode.

### CODE REVIEW

File: src/components/tutorials/common/technical-details-pane.jsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import { Tabs } from 'react-tabs';
+import PropTypes from 'prop-types';
```
Comment (samskeller) [inline src/components/tutorials/common/technical-details-pane.jsx]: Not sure this is a necessary import...?

Comment (samskeller) [inline src/components/tutorials/common/technical-details-pane.jsx]: Not sure this is a necessary import?

**Comment by kencheeto** [comment]
crud! i didn't see this pr, i ended up doing a lot of the same thing in https://github.com/CMSgov/qpp-submissions-docs/pull/62 :( we could merge this first and I can fix merge conflicts in my pr

**Comment by kencheeto** [comment]
subsumed by #62 , closing

**Pull request closed without merging**